### ISSUE TITLE
[Proposal] v1 Release Documentation ToC

### ISSUE THREAD

**Issue by EisenbergEffect** [microsoft/fast-dna#3043]
## Description

A proposal for the necessary documentation content needed for v1 release.

## ToC

- [x] Introduction
- Using FAST Components
    - [x] Getting Started
    - Components
        - [x] fast-anchor
        - [x] fast-badge
        - [x] fast-button
        - [x] fast-card
        - [x] fast-checkbox
        - [x] fast-design-system-provider
        - [x] fast-dialog
        - [x] fast-divider
        - [x] fast-flipper
        - [x] fast-progress
        - [x] fast-radio-group
        - [x] fast-radio
        - [x] fast-slider-label
        - [x] fast-slider
        - [x] fast-switch
        - [x] fast-tab
        - [x] fast-text-area
        - [x] fast-text-field
    - [x] Styling Components
    - Integrations
        - [x] ASP.NET
        - [x] Aurelia
        - [x] Blazor
        - [x] TypeScript and Webpack
             - [ ] Template string minification (CSS and HTML)
        - more...
- Using FAST Element
    - [x] Getting Started
    - [x] Defining Elements
    - [x] Declaring Templates
    - [x] Using Directives
    - [x] Observables and State
    - [x] Working with Shadow DOM
    - [x] Leveraging CSS
    - [x] Tooling
- API Reference
    - [x] fast-foundation
    - [x] fast-components
    - [x] fast-components-msft
    - [x] fast-element
- Community Contribution
    - [x] Joining the Community
    - [x] Code of Conduct
    - [x] Contributor Guide
    - [x] Building and Testing
    - [x] Writing Documentation
- Resources
    - [x] FAQ
    - [x] Browser Support
    - [x] Glossary

### Notes

* The "Introduction" and all content in the sections "Using FAST Components", "Using FAST Element", "Community Contributions", and "Resources" will be hand written using Markdown.
* Most of the content for "Using FAST Element" is already written.
* Under "Using FAST Components/Components" we should have a page for each component we ship. We can take some of the content from our specs and some of the demos from our storybook fixtures. 
* Under "Integrations" we should have docs on how to get started with as many frameworks as possible. I've listed a couple that have already been tested or are in the process of being tested, so I think they are good candidates for launch. If we can't get this in place on day one, we can easily add this in the following weeks. I think it will be valuable for people to know how to get the components working in their existing contexts though.
* The "API Reference" section will be generated. We'll take d.ts files as input and use TypeDoc to spit out a JSON file. We'll then write a custom script to take the JSON as input and generate a Docusaurus page for each library.

**Comment by awentzel** [comment]
In particular,

> Under "Integrations" we should have docs on how to get started with as many frameworks as possible. I've listed a couple that have already been tested or are in the process of being tested, so I think they are good candidates for launch. If we can't get this in place on day one, we can easily add this in the following weeks. I think it will be valuable for people to know how to get the components working in their existing contexts though.

And Getting Started with something similar to Create React App will streamline getting up and running.  This will remove all barriers to entry so it's easy.  If not easy, we have more work to do.

**Comment by EisenbergEffect** [comment]
I don't think we'll be able to create a CLI before our launch. So, I'm proposing a few tutorials on how best to set it up with particular frameworks. I'm not convinced a CLI would handle that well or if we want to either. Lots of code to integrate with tons of frameworks and setups can be difficult to maintain. That's why I'm leaning to a doc that can easily be updated even by the respective communities. I bet I could get the Aurelia team to write the docs for us, for example.

**Comment by nicholasrice** [comment]
@EisenbergEffect have you written scripts to extract APIs docs from TypeDoc JSON? In my experience it hasn't been straight-forward but I very-well could have been approaching it incorrectly. If not, we may want to consider just using TypeDoc site-generator (maybe with a custom theme?) for the short term.

**Comment by EisenbergEffect** [comment]
@nicholasrice I have some existing code where we used the json output of typedocs and generated custom pages from that. I did that for Aurelia so I think I can whip something up for us as well. I just haven't gotten a chance to actually do it yet 😢 It's the next thing on my docs todo list.

**Comment by nicholasrice** [comment]
Okay cool. Otherwise I think this looks great!

**Comment by chrisdholt** [comment]
@EisenbergEffect @awentzel We've talked about the CLI but I don't think the intent is to have it be a WC create-react-app like thing. Our initial discussion on the CLI was to create your own design system and I believe that still has value, though agree it may be difficult before launch. I think it needs to be on our roadmap for sure.

In the same vein, @Falkicon and I discussed getting the `fast-frame` UI system put together which would be the default system shown in the CLI. I think having that prior to launch would be a value-add and help serve as an example - it can be in a separate repo as well that folks can clone to get started prior to having the CLI.

**Comment by EisenbergEffect** [comment]
I'm just guarding any expansion to scope right now since I know already I'm not going to get nearly as many tests in place as I'd like before launch, for example. And, actually writing the above docs will take a good bit of time 😄 

I wanted to drop a proposed ToC here so we could possibly divvy up the fast components docs among the component authors next week maybe and knock those out.

**Comment by chrisdholt** [comment]
@EisenbergEffect certainly - I don't think it's related to the docs, but I did want to bring clarity on the goal of the CLI we've previously discussed.

**Comment by awentzel** [comment]
I'm not suggesting an actual "CLI" or specific technology, only that covering the basics first is important.  Design systems are an advanced topic, and I think the early adopters first will want to just get up and running with ease. Then inch towards advanced topics.  Totally agree with @EisenbergEffect scope management.

**Comment by EisenbergEffect** [comment]
@chrisdholt One thing I didn't outline was what we want each of the component doc pages to look like. I can take a stab at proposing that, but I thought I'd ask if there's anyone on your team that might be interested to make an initial proposal on that. I can happily meet with someone and do a quick brain dump in about 5 min or so. I think it would be nice to have someone take a component and build out a page that others can follow.

**Comment by janechu** [comment]
A very minor issue but one that I don't want to lose for documentation is that there is a brief moment where content is un-styled, I mitigated this with a suggestion from @nicholasrice :

```css
:not(:defined) {
    visibility: hidden;
}
```

**Comment by EisenbergEffect** [comment]
I've updated the proposed ToC and added checkboxes to track progress of each doc (or code gen output).

Any thoughts on what we should definitely include in our FAQ?

**Comment by janechu** [comment]
Just a few suggestions:
- Why are you using web components?
- Can I use other component libraries with this one?
- How can I get involved with the FAST project?

Outside of that I'm sure once we go public and issues start getting filed.

**Comment by janechu** [comment]
One other thing is we might cross link to various frameworks like React on how they can integrate web components if there is some extra work that needs to be done.

**Comment by nicholasrice** [comment]
I noticed that the minification story is a made complicated by the use of template strings for HTML and CSS so I've added section to address it in documentation. Template strings are not minified by default in the webpack production builds we're using.

Minification of HTML templates *could* have the potential to create runtime bugs so this may be something we want to test.

CSS should be able to be minified safely.
